generated/nidcpower/nidcpower/session.py

```diff
@@ -4801,11 +4801,39 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
+        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance', 'channel'])
 
         voltage_measurements, current_measurements, in_compliance = self._fetch_multiple(timeout, count)
 
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=in_compliance[i]) for i in range(count)]
+        if self._repeated_capability == '':
+            channel_names = self._get_channel_names(range(self.channel_count))
+        else:
+            # Check if the session was opened with independent_channels set to True by checking if
+            #  _get_channel_name() returns channel names with prefix
+            first_channel_name = self._get_channel_name(1)
+            if '/' in first_channel_name:
+                # If there is any repeated capabilities without prefix, the session must have only
+                # one instrument, so just get the prefix from the first channel and add to all of
+                # the repeated capabilities that are without prefix
+                default_prefix = first_channel_name[:first_channel_name.find('/') + 1]
+                channel_names = _converters.convert_independent_channels_repeated_capabilities(
+                    self._repeated_capability,
+                    default_prefix
+                )
+            else:
+                channel_names = _converters.convert_repeated_capabilities(
+                    self._repeated_capability,
+                    ''
+                )
```

Comment:
Move all of this logic into `_converters`. Better than having some channel parsing logic there and some here.

Reply:
This will also avoid code duplication between this method and `measure_multiple`.

Reply:
Thanks for the suggestion. I have moved the logic to `convert_channels_repeated_capabilities()` inside the converters.py in my latest commit.